src/App/Pages/Vault/ViewPage.xaml

```diff
@@ -178,29 +179,49 @@
                                 <controls:MonoLabel
                                     Text="{Binding TotpCodeFormatted, Mode=OneWay}"
+                                    IsVisible="{Binding ShowUpgradePremiumTotpText, Converter={StaticResource inverseBool}}"
                                     StyleClass="box-value"
                                     Grid.Row="1"
-                                    Grid.Column="0" />
+                                    Grid.Column="0"
+                                    VerticalTextAlignment="Start"
+                                    VerticalOptions="Start" />
+                                <controls:CircularProgressbarView
+                                    Progress="{Binding TotpProgress}"
+                                    Grid.Row="0"
+                                    Grid.Column="1"
+                                    Grid.RowSpan="2"
+                                    HorizontalOptions="FillAndExpand"
+                                    VerticalOptions="FillAndExpand" />
                                 <Label
                                     Text="{Binding TotpSec, Mode=OneWay}"
                                     Style="{DynamicResource textTotp}"
-                                    Margin="0, 0, 10, 0"
                                     Grid.Row="0"
                                     Grid.Column="1"
                                     Grid.RowSpan="2"
-                                    HorizontalOptions="End"
-                                    HorizontalTextAlignment="End"
-                                    VerticalOptions="CenterAndExpand" />
-                                <controls:IconButton 
+                                    StyleClass="text-sm"
+                                    VerticalTextAlignment="Center"
+                                    HorizontalTextAlignment="Center"
+                                    HorizontalOptions="FillAndExpand"
+                                    VerticalOptions="FillAndExpand" />
+                                <controls:IconButton
                                     StyleClass="box-row-button, box-row-button-platform"
                                     Text="{Binding Source={x:Static core:BitwardenIcons.Clone}}"
                                     Command="{Binding CopyCommand}"
+                                    IsVisible="{Binding CanAccessPremium}"
                                     CommandParameter="LoginTotp"
                                     Grid.Row="0"
                                     Grid.Column="2"
                                     Grid.RowSpan="2"
                                     AutomationProperties.IsInAccessibleTree="True"
                                     AutomationProperties.Name="{u:I18n CopyTotp}" />
+                                <Label
+                                    Text="{u:I18n UpgradeToPremiumTotpText}"
```

Comment:
This is probably showing nothing. It should be `{u:I18n PremiumSubscriptionRequired}`